Code/DistGeom/DistGeomUtils.cpp

```diff
@@ -237,5 +242,209 @@ namespace DistGeom {
       }
     }
     return field;
-  }
+  } // constructForceField
+
+  ForceFields::ForceField *construct3DForceField(const BoundsMatrix &mmat,
+                                                 RDGeom::Point3DPtrVect &positions,
+                                                 const std::vector<std::pair<int, int> > &bonds,
+                                                 const std::vector<std::vector<int> > &angles,
+                                                 const std::vector<std::vector<int> > &expTorsionAtoms,
+                                                 const std::vector<std::pair<std::vector<int>, std::vector<double> > > &expTorsionAngles,
+                                                 const std::vector<std::vector<int> > &improperAtoms,
+                                                 const std::vector<int> &atomNums) {
+    unsigned int N = mmat.numRows();
+    CHECK_INVARIANT(N == positions.size(), "");
+    CHECK_INVARIANT(expTorsionAtoms.size() == expTorsionAngles.size(), "");
+    ForceFields::ForceField *field = new ForceFields::ForceField(positions[0]->dimension());
+    for (unsigned int i = 0; i < N; ++i){
+      field->positions().push_back(positions[i]);
+    }
+
+    // keep track which atoms are 1,2- or 1,3-restrained
+    std::vector<std::vector<int> > atomPairs(N, std::vector<int>(N, 0));
```

Comment:
Couldn't this just be a boost::dynamic_bitset<> that is N*N long?
You could then just set values with: `atomPairs[i*N+j] = 1`